### PR TITLE
[node-manager] add docs for static provider

### PR DESCRIPTION
**Pull request by artbin**
## Description

Add documentation for Cluster API Provider Static.

## Why do we need it in the patch release (if we do)?

Cluster API Provider Static is ready for use, and the documentation is very much needed.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: node-manager
type: feature
summary: Add documentation for Cluster API Provider Static.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'deploy/web/stage'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'deploy/web/stage'...

File: modules/040-node-manager/crds/sshcredentials.yaml [hunk 1]
```diff
@@ -38,17 +39,27 @@ spec:
               description: SSHCredentialsSpec defines the desired state of SSHCredentials
               properties:
                 privateSSHKey:
+                  description: |
+                    Private SSH key in PEM format encoded as base64 string.
                   type: string
                 sshExtraArgs:
+                  description: |
+                    A list of additional arguments to pass to the openssh command.
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/crds/sshcredentials.yaml:47]: I think there is should be some examples

File: modules/040-node-manager/crds/sshcredentials.yaml [hunk 2]
```diff
@@ -38,17 +39,27 @@ spec:
...
                 sshPort:
+                  description: |
+                    A port to connect to the host via SSH.
                   default: 22
                   maximum: 65535
                   minimum: 1
                   type: integer
                 sudoPassword:
+                  description: |
+                    A password for the user to use sudo.
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/crds/sshcredentials.yaml:58]: may be sudo password ?

File: modules/040-node-manager/crds/sshcredentials.yaml [hunk 3]
```diff
@@ -38,17 +39,27 @@ spec:
...
                   type: string
                 user:
+                  description: |
+                    A name of the user to connect to the host via SSH.
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/crds/sshcredentials.yaml:62]: username ?

File: modules/040-node-manager/crds/staticcluster.yaml [hunk 4]
```diff
@@ -58,10 +58,20 @@ spec:
               description: StaticClusterStatus defines the observed state of StaticCluster
               properties:
                 failureMessage:
+                  description: |
+                    FailureMessage will be set in the event that there is a terminal problem
+                    reconciling the OpenStackCluster and will contain a more verbose string suitable
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/crds/staticcluster.yaml:63]: OpenStackCluster is wrong here

File: modules/040-node-manager/crds/staticcluster.yaml [hunk 5]
```diff
@@ -58,10 +58,20 @@ spec:
...
+                    for logging and human consumption.
                   type: string
                 failureReason:
+                  description: |
+                    FailureReason will be set in the event that there is a terminal problem
+                    reconciling the OpenStackCluster and will contain a succinct value suitable
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/crds/staticcluster.yaml:69]: same

File: modules/040-node-manager/crds/staticinstance.yaml [hunk 6]
```diff
@@ -33,7 +33,7 @@ spec:
       name: v1alpha1
       schema:
         openAPIV3Schema:
-          description: StaticInstance is the Schema for the staticinstances API
+          description: StaticInstance is the Schema for the Cluster API Provider Static
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/crds/staticinstance.yaml:36]: i think this is not a schema for CAPS. This is a way to describe static machines in the cluster

File: modules/040-node-manager/docs/EXAMPLES.md [hunk 7]
```diff
@@ -110,3 +110,169 @@ spec:
...
+  name: worker-4cpu-16gb
+spec:
+  nodeType: Static
+  staticInstances:
+    count: 2
+    labelSelector:
+      matchLabels:
+        cpu: 4
+        memory: 16GB
+apiVersion: deckhouse.io/v1
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/docs/EXAMPLES.md:217]:
```suggestion
        memory: 16GB
---       
apiVersion: deckhouse.io/v1
```

File: modules/040-node-manager/docs/FAQ.md [hunk 8]
```diff
@@ -945,4 +954,27 @@ metadata:
     werf.io/track-termination-mode: NonBlocking
 ```
 
+## How do I manually clean up a static node managed by CAPS?
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/docs/FAQ.md:957]: We can clean up not only CAPS-managed nodes

**Review by RomanenkoDenys — COMMENTED** [review]
some comments

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

File: modules/040-node-manager/crds/sshcredentials.yaml (hunk 2 above)
Comment (z9r5) [reply modules/040-node-manager/crds/sshcredentials.yaml:60]: Updated

File: modules/040-node-manager/crds/sshcredentials.yaml (hunk 3 above)
Comment (z9r5) [reply modules/040-node-manager/crds/sshcredentials.yaml:64]: Updated

File: modules/040-node-manager/crds/sshcredentials.yaml (hunk 1 above)
Comment (z9r5) [reply modules/040-node-manager/crds/sshcredentials.yaml:49]: @artbin can you add some examples in the comment?

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

File: modules/040-node-manager/crds/staticinstance.yaml (hunk 6 above)
Comment (z9r5) [reply modules/040-node-manager/crds/staticinstance.yaml]: Updated

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

File: modules/040-node-manager/crds/sshcredentials.yaml (hunk 1 above)
Comment (artbin) [reply modules/040-node-manager/crds/sshcredentials.yaml:49]: done

**Review by kirkonru — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Comment by z9r5** [comment]
Cherry-picked into release-1.54

**Pull request merged**